app/models/agents/http_status_agent.rb

```diff
@@ -46,29 +50,47 @@ def validate_options
       errors.add(:base, "a url must be specified") unless options['url'].present?
     end
 
+    def header_array(str)
+      str.split(',').map { |el| el.strip }
+    end
+
     def check
-      check_this_url interpolated[:url]
+      check_this_url interpolated[:url], header_array(interpolated[:headers_to_save])
```

Comment:
Since `interpolated[:headers_to_save]` could be `null`, I'd suggest changing this to

`header_array(interpolated[:headers_to_save] || '')`

Or putting that in the `header_array` method, like

```ruby
def header_array(str)
  (str || '').split(',').map(&:strip)
end
```